cmd/device-hub/main.go

```diff
@@ -33,6 +33,8 @@ type config struct {
 	CertFile   string `envconfig:"TLS_CERT_FILE"`
 	KeyFile    string `envconfig:"TLS_KEY_FILE"`
 	Data       string `envconfig:"DATA" default:"."`
+	LogFile    string `envconfig:"CONFIG_FILE" default:""`
```

Comment:
I think the envconfig should be LOG_FILE.

There should be a default for the log file even if it is a file in the local folder.

I do understand that you are using an empty log file setting to switch on whether to write to a log file or stdout - please see later comment.